serverless/architecture/serverless-serving-architecture.adoc

```diff
@@ -0,0 +1,29 @@
+include::modules/serverless-document-attributes.adoc[]
+[id="serverless-serving-architecture"]
+= Knative Serving architecture
+:context: serverless-serving-architecture
+include::modules/common-attributes.adoc[]
+
+toc::[]
+
+Knative Serving supports deploying and managing serverless applications, by providing a set of objects as Kubernetes CustomResourceDefinitions (CRDs) that define and control the behavior of serverless workloads on an {product-title} cluster.
+
+Developers use these CRDs to create custom resource instances that can be used as building blocks to address complex use cases. For example:
+
+* Rapidly deploying serverless containers.
+* Automatically scaling pods.
+// TODO: Add other use cases, more advanced tutorials?
+
+[id="serverless-serving-architecture-custom-resources"]
+== Knative Serving CRDs
+// TODO: Request a good Serving arch diagram
+
+Service:: The `service.serving.knative.dev` CRD automatically manages the life cycle of your workload to ensure that the application is deployed and reachable through the network. It creates a Route, a Configuration, and a new Revision for each change to a user created Service, or custom resource.
++
+Most developer interactions in Knative are carried out by modifying Services, which simplifies Kubernetes operations, as this resource can be viewed as a replacement for a Route-Service-Deployment mechanism with a built-in autoscaler.
```

Comment:
[Route-Service-Deployment [1]](https://docs.openshift.com/container-platform/3.6/dev_guide/deployments/advanced_deployment_strategies.html) isn't a mechanism. We can't say that. It's a Kubernetes or rather Openshift pattern, a good practice. Something that nearly everybody use. In Kubernetes it's [Ingress-Service-Deployment [2]](https://medium.com/@dwdraju/how-deployment-service-ingress-are-related-in-their-manifest-a2e553cf0ffb).

[1] https://docs.openshift.com/container-platform/3.6/dev_guide/deployments/advanced_deployment_strategies.html
[2] https://medium.com/@dwdraju/how-deployment-service-ingress-are-related-in-their-manifest-a2e553cf0ffb

Reply:
I would not give it a name like that at all, that doesn't really help anybody imo. At least i'm not aware of expressions like `Route-Service-Deployment` being commonly used.

Instead, let's just say that a service unifies the creation, management and scaling of a deployment, and makes sure that the pods of said deployment are reachable via HTTP.

Reply:
Cool, changing it to pattern.

Reply:
Sorry Markus just saw this comment for some reason. I'll remove this.

@cardil this is probably a good lesson to be honest. Once something is in QE review for docs, we should be reviewing for the accuracy of the info in the doc **only**, unless it's vital that we add more info for some reason. This is not the time to request a bunch of additional content or changes that aren't really necessary or could be followed up by another, more relevant, PR.

Reply:
Maybe I should prepare you the suggestions I've done. My original intent here was just that - "accuracy of the info in the doc". Original PR listed those CRs without saying that users should change only Service, and others are read only.

Reply:
Following this up in Slack